From a2ccc325b7b4ef2f22cf98b6fcdb5ca16e383e66 Mon Sep 17 00:00:00 2001
From: Todd Whiteman <todd.whiteman@joyent.com>
Date: Tue, 4 Jun 2019 15:23:27 -0700
Subject: [PATCH] TRITON-1722 docker volume creation failure test is failing in
 nightly Reviewed by: Orlando Vazquez <orlando@joyent.com> Approved by:
 Orlando Vazquez <orlando@joyent.com>

---
 ...li-nfs-shared-volumes-create-failure.test.js | 17 +++++------------
 1 file changed, 5 insertions(+), 12 deletions(-)

diff --git a/test/integration/cli-nfs-shared-volumes-create-failure.test.js b/test/integration/cli-nfs-shared-volumes-create-failure.test.js
index e3fe90f..9d4a547 100644
--- a/test/integration/cli-nfs-shared-volumes-create-failure.test.js
+++ b/test/integration/cli-nfs-shared-volumes-create-failure.test.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2019, Joyent, Inc.
  */
 
 //
@@ -107,21 +107,14 @@ test('Volume creation should fail when provision fails', function (tt) {
                 size: '10G',
                 name: testVolumeName
             }, function volumeCreated(err, stdout, stderr) {
-                var expectedErr = 'Error response from daemon: (InternalError) '
-                    + 'volume creation failed';
-                var matches;
-
-                // Make a RegExp from the expectedErr but we need to escape the
-                // '(' and ')' characters to '\(' and '\)' so that the regex
-                // will not treat that as a grouping.
-                var re = new RegExp(expectedErr.replace(/[()]/g, '\\$&'));
-
-                matches = stderr.match(re);
+                var expectedErr = 'Error response from daemon: '
+                    + 'problem creating volume';
+                var matches = stderr.match(expectedErr);
 
                 t.ok(err, 'volume creation should not succeed');
                 // with this, we get the actual error message if it fails
                 t.equal((matches ? matches[0] : stderr), expectedErr,
-                    'expected InternalError');
+                    'expected "problem creating volume" error message');
 
                 t.end();
             });
-- 
2.21.0

